docs/types/literal-suffixes.rst

```diff
@@ -0,0 +1,243 @@
+| ``address``                                                 | - :ref:`Address literals<address_literals>`                    |
++-------------------------------------------------------------+----------------------------------------------------------------+
+| ``bytes1``, ..., ``bytes32``                                | - Hexadecimal number literals (not for ``bytes20``)            |
+|                                                             | - :ref:`Hexadecimal string literals<hexadecimal_literals>`     |
+|                                                             | - :ref:`String literals<string_literals>`                      |
+|                                                             | - :ref:`Unicode literals<unicode_literals>`                    |
+|                                                             | - :ref:`Zero<rational_literals>`                               |
++-------------------------------------------------------------+----------------------------------------------------------------+
+| ``bytes``                                                   | - :ref:`Hexadecimal string literals<hexadecimal_literals>`     |
+|                                                             | - :ref:`String literals<string_literals>`                      |
+|                                                             | - :ref:`Unicode literals<unicode_literals>`                    |
++-------------------------------------------------------------+----------------------------------------------------------------+
+| ``string``                                                  | - :ref:`String literals<string_literals>`                      |
+|                                                             | - :ref:`Unicode literals<unicode_literals>`                    |
++-------------------------------------------------------------+----------------------------------------------------------------+
+
+For two-parameter suffix functions, the first parameter (representing the mantissa) can be of any integer type.
+The second parameter (the exponent) must be of an unsigned integer type.
+
+.. note::
+    40-digit literals prefixed with ``0x`` such as, for example, ``0xdCad3a6d3569DF655070DEd06cb7A1b2Ccd1D3AF``
+    always represent ``address`` literals in the language.
+    To invoke a suffix accepting ``bytes20`` you must use one of the other literal kinds implicitly
+    convertible to ``bytes20``, e.g., a hexadecimal string literal
+    (``hex"dCad3a6d3569DF655070DEd06cb7A1b2Ccd1D3AF"``).
+
+.. note::
+    Suffix functions accepting ``address payable`` are not allowed since address literals are never payable.
+
+Suffix functions may not accept or return reference types with ``storage`` or ``calldata`` locations.
+
+.. index:: function;call
+.. _calling_suffix_functions:
+
+Calling Suffix Functions
+------------------------
+
+There are two ways to call a suffix function:
+
+#. Suffix call.
+#. Function call.
```

Comment:
No, this was meant to be just a summary, spelling out the options before we delve into them. What you propose looks would be more like a table of contents and it seems a bit out of place to me. Especially since you have links to sections in the menu on the left.

Actually you may have a point that it's already a bit too much like a TOC. Maybe it would be different if there was some text describing each option. I reworded it so that now it's just a sentence. Is that better?

I could still make them links, but this looks as is I wanted you to immediately jump forward to those sections, while I think it's better if you read them in order.